crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs

```diff
@@ -128,7 +167,9 @@ impl SenderData {
 
     /// Returns true if this is SenderKnown and `master_key_verified` is true.
```

Comment:
It looks like this comment was wrong before, but definitely needs updating now.

Reply:
Looking at this function, it looks like it's only used in OlmMachine to determine whether it should recalculate the sender_data.  So I definitely don't want it to return `true` if it's `SenderUnverifiedButPreviouslyVerified`.  But I'm not sure that it should return `true` if it's `SenderUnverified` either, although that would match the previous behaviour, since `sender_data_to_verification_state` returns different values if the sender is verified or not.  What do you think?  What was your rationale for having it return true when the sender was unverified?

Reply:
I'm happy for you to merge this as-is or with one of the changes I mention above. If you want to change the behaviour I'll want to discuss it further.

Reply:
> Maybe this function should be moved inside `OlmMachine` and be named `should_recalculate`?

Done.  It's ended up adding a bunch of negations, which I'm not really a fan of, but to avoid the negations, I'd either need to rename to `should_not_recalculate`, or switch the logic in the place where it's called, both of which seem at least as bad.

If you're OK with the way it is, can you hit the merge button?  I don't have permissions.